### PR TITLE
Reference masking prior to pmdtools, when requested.

### LINKED ISSUE #829: Update sequenceTools
Hi team,

I've updated the sequenceTools package on bioconda a while ago to version 1.5.1, which among other improvements now has Plink output support. Please update the version shipped with the nf-core/eager pipeline. 

Thank you!

### LINKED ISSUE #844: PMDTools with masked reference is borked
I noticed PMDtools would have a similar issue as in #838 , but when fixing it I was told by @TCLamnidis my 'fix' didn't make sense, but I also didn't actually know what it was doing :sweat_smile:. Furhter investigation by @TCLamnidis showed the masked reference didn't even work in this case due to a missing flag. He will investigate and fix (I can't describe fully as I still don't understand :sweat_smile: )

### PR DESCRIPTION
**Pull request by TCLamnidis**
## Changes

- Added masking of reference fasta prior to running pmdtools, when requested.
  - Updated explanation of `pmdtools_reference_mask` parameter in schema.
- Bumped sequencetools `1.4.0.5` -> `1.5.2`

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.

Closes #844 
Closes #829

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -2127,6 +2133,33 @@ process mapdamage_rescaling {
...
 
+process mask_reference_for_pmdtools {
+    label 'sc_tiny'
+    tag "${fasta}"
+    publishDir "${params.outdir}/reference_genome/masked_reference", mode: params.publish_dir_mode
+
+    when: (params.pmdtools_reference_mask && params.run_pmdtools)
+
+    input:
+    file fasta from ch_unmasked_fasta_for_masking.collect()
```
Comment (jfy133) [inline main.nf:2144]: Do you really need `collect` here? You would only need to run this process once, right?

**Review by jfy133 — COMMENTED** [review]
Code looks good, so just waiting for the 'fluff' ;) <3

File: main.nf (hunk 1 above)
Comment (TCLamnidis) [reply main.nf:2144]: Must have been leftover from copy-pasting the fasta input of the `pmdtools` process. Will remove.

**Comment by TCLamnidis** [comment]
PR seems stuck. will recreate to force PR update.

**Pull request closed without merging**